cloud-platform-po-linter.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/prometheus-operator/prometheus-operator/pkg/admission"
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus/prometheus/model/rulefmt"
+	githubaction "github.com/sethvargo/go-githubactions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func prDir(token, githubrepo, githubref string) ([]string, error) {
+	if token == "" {
```

Comment:
While we're being defensive here (which is good) - what happens if we pass empty `gitHubRef` and `repo` strings?

Reply:
Also, can I do annon api calls?